metrics/ci_integrations/test/destination/firestore/config/validation_delegate/firestore_destination_validation_delegate_test.dart

```diff
@@ -68,16 +106,80 @@ void main() {
       return when(firebaseAuth.signIn(withEmail, withPassword));
     }
 
+    PostExpectation<Firestore> whenCreateFirestore() {
+      whenSignIn().thenAnswer((_) => Future.value(user));
+
+      return when(firestoreFactory.create(projectId, firebaseAuth));
+    }
+
+    PostExpectation<CollectionReference> whenReferenceCollection({
+      String withName,
+    }) {
+      whenCreateFirestore().thenReturn(firestore);
+
+      return when(firestore.collection(withName));
+    }
+
+    PostExpectation<Future<List<Document>>> whenReadDocuments() {
+      whenReferenceCollection(
+        withName: projectsCollectionName,
+      ).thenReturn(collection);
+
+      return when(collection.getDocuments());
+    }
+
+    PostExpectation<DocumentReference> whenReferenceDocument({
+      String withName,
+    }) {
+      whenReferenceCollection(
+        withName: projectsCollectionName,
+      ).thenReturn(collection);
+
+      return when(collection.document(withName));
+    }
+
+    PostExpectation<Future<Document>> whenGetDocument() {
```

Comment:
```suggestion
    PostExpectation<Future<Document>> whenGetMetricsProject() {
```
Would it improve readability?